crawler/src/main/java/com/javaacademy/crawler/jsoup/MatrasScrapper.java

```diff
@@ -108,43 +99,20 @@ String getSmallThumbnail() {
     @Override
     double getListPrice() {
         Elements elements = getDoc().select("div.old-price");
-        if (elements.isEmpty()) {
-            return 0;
-        } else {
-            double price = 0;
-            try {
-                price = Double.parseDouble(elements.text().replace("\u00a0zł", "").replace(",", "."));
-            } catch (NumberFormatException e) {
-                AppLogger.logger.log(Level.WARNING, "Exception while parsing, ", e);
-            }
-            return price;
-        }
+        return elements.isEmpty() ? 0 :
+                parsePrice(elements.text().replace("\u00a0zł", "").replace(",", "."));
     }
 
     @SuppressWarnings("Duplicates")
     @Override
     double getRetailPrice() {
         Elements elements = getDoc().select("div.this-main-price");
-        if (elements.isEmpty()) {
-            return 0;
-        } else {
-            double price = 0;
-            try {
-                price = Double.parseDouble(elements.text().replace("\u00a0zł", "").replace(",", "."));
-            } catch (NumberFormatException e) {
-                AppLogger.logger.log(Level.WARNING, "Exception while parsing, ", e);
-            }
-            return price;
-        }
+        return elements.isEmpty() ? 0 :
+                parsePrice(elements.text().replace("\u00a0zł", "").replace(",", "."));
```

Comment:
.attr("content") gives you number without unit and with .

Reply:
:+1: